### PR TITLE
test(backend): fix flaky backend test

### PR DESCRIPTION
**Pull request by Linchin**
**Description of your changes:**
Since #9059, the backend test has been rather flaky. After some investigation, it turned out that the tasks inside the root dag had a random order, and an added unit test in #9059 exposed this issue. This PR fixed the issue by ordering the tasks alphabetically.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/v2/compiler/argocompiler/dag.go [hunk 1]
```diff
@@ -48,7 +49,18 @@ func (c *workflowCompiler) DAG(name string, componentSpec *pipelinespec.Componen
...
-	for taskName, kfpTask := range dagSpec.GetTasks() {
+	tasks := dagSpec.GetTasks()
+	// Iterate through tasks in deterministic order to facilitate testing.
+	// Note, order doesn't affect compiler with real effect right now.
+	// In the future, we may consider using topology sort when building local
+	// executor that runs on pipeline spec directly.
+	keys := make([]string, 0, len(tasks))
+	for key := range tasks {
+		keys = append(keys, key)
+	}
```
Comment (chensun) [inline backend/src/v2/compiler/argocompiler/dag.go:60]: I think this chunk can be just

```golang
keys := dagSpec.GetTasks().Keys()
```

Comment (chensun) [reply backend/src/v2/compiler/argocompiler/dag.go:60]: My bad, ignore me then.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks for the quick fix.

File: backend/src/v2/compiler/argocompiler/dag.go (hunk 1 above)
Comment (Linchin) [reply backend/src/v2/compiler/argocompiler/dag.go:60]: With 
> ```go
> keys := reflect.ValueOf(dagSpec.GetTasks()).MapKeys()
> ```
It doesn't feel very readable? it also introduces a new package. The variable `tasks` is used again at line #58 so I don't think it makes a huge difference.

**Pull request merged**